components/net/protocols/data.rs

```diff
@@ -5,71 +5,19 @@
 use std::future::Future;
 use std::pin::Pin;
 
-use data_url::forgiving_base64;
-use headers::{ContentType, HeaderMapExt};
+use data_url::DataUrl;
+use headers::HeaderValue;
 use http::StatusCode;
-use mime::Mime;
 use net_traits::request::Request;
 use net_traits::response::{Response, ResponseBody};
 use net_traits::{NetworkError, ResourceFetchTiming};
-use percent_encoding::percent_decode;
-use servo_url::ServoUrl;
-use url::Position;
 
 use crate::fetch::methods::{DoneChannel, FetchContext};
 use crate::protocols::ProtocolHandler;
 
 #[derive(Default)]
 pub struct DataProtocolHander {}
 
-enum DecodeError {
-    InvalidDataUri,
-    NonBase64DataUri,
-}
-
-type DecodeData = (Mime, Vec<u8>);
-
-fn decode(url: &ServoUrl) -> Result<DecodeData, DecodeError> {
-    // data_url could do all of this work for us,
-    // except that it currently (Nov 2019) parses mime types into a
-    // different Mime class than other code expects
```

Comment:
Is this still a problem? Relevant issue is still open: https://github.com/servo/servo/issues/2909

Reply:
I think the problem from #2909 still applies. We parse mime using data_url crate then write it as string to header in response that gets parsed again (using different parser), but similar jumping was also done before, with `mime` crate doing mime parsing. Maybe we should just keep mime as str and avoid parsing it at this stage? I am not really sure, I need to dig into spec more. CC @SimonSapin

Reply:
I have no memory of the details here, sorry!

Reply:
I reread all issues again and I think I misunderstand the situation.